[docs] Reword palette intention and fix format (#21477)
mui-org/material-ui: docs/src/pages/customization/palette/palette.md

## Palette colors

> A color is a mapping of a palette to a given meaning within your application.

Reviewer: I don't think that it's clearer, I don't understand what the first nor the second sentence mean. Maybe

```suggestion
A "palette color" is a mapping of colors to a given intention within your application.
```

> A "palette color" is a mapping of colors to a given intention within your application.

Reviewer 2: I thought we decided to drop this sentence?

Author: So we won't have this one at all? I'm confused :(

{{"demo": "pages/customization/palette/Intentions.js", "bg": "inline", "hideToolbar": true}}

> The default palette uses the shades prefixed with `A` (`A200`, etc.) for the secondary color,

Reviewer: ```suggestion
The default palette uses the shades prefixed with `A` (`A200`, etc.) for the secondary palette color,
```

> and the un-prefixed shades for the other colors.

Reviewer: ```suggestion
and the un-prefixed shades for the other palette colors.
```

### Using a color object

> The simplest way to customize a color object is to import one or more of the provided colors

Reviewer: ```suggestion
The simplest way to customize a palette color is to import one or more of the provided colors
```

> and apply them to a palette:

Reviewer: ```suggestion
and apply them:
```

### Providing the colors directly

> If you wish to provide more customized colors, you can either create your own palette color,

Reviewer 2: ```suggestion
If you wish to provide more customized colors, you can either create your own color object,
```

I may be mistaken, but I think that one was correct.

Author: Actually i want to change the next sentence, somehow it got omitted. 
Do you have any suggestions on this one? @mbrookes 
```
or directly supply colors to some or all of the intention's keys:
```

Reviewer 2: or directly supply colors to some or all of the `theme.palette` keys:

?

Reviewer: I have no specific preferences. I feel that if we propose custom keys for the second option, we might as well propose the most structured structure for the first one: palette color, which implies the underlying type.

Reviewer 3: Hmm, I find that color object makes sense to me but after reading the page and seeing that we describe the color objects as palette colors we should stay consistent and use "palette colors". 

> palette color, which implies the underlying type.

The underlying type is an object with colours as the values. We even reference this in one of the titles: "Using a color object".

> or directly supply colors to some or all of the `theme.palette`'s keys:

Reviewer 2: ```suggestion
or directly supply colors to some or all of the `theme.palette` keys:
```

### Providing the colors directly

> If you wish to provide more customized colors, you can either create your own color object,

Reviewer: ```suggestion
If you wish to provide more customized colors, you can either create your own palette color,
```

> or directly supply colors to some or all of the color's keys:

Reviewer: ```suggestion
or directly supply colors to some or all of the `theme.palette`'s keys:
```

Author: this one makes more sense !

});
```

> As in the example above, if the color object contains custom colors using any of the

Reviewer: ```suggestion
As in the example above, if the palette color contains custom colors using any of the
```
